Sources/MessagingPushAPN/RichPush/MessagingPush+NotificationServiceExtension.swift

```diff
@@ -40,6 +42,18 @@ public extension MessagingPush {
             return false
         }
 
+        // If cioHandleBadge exists and value is true
+        if let cioHandleBadge = pushContent.cioHandleBadge, cioHandleBadge  {
+            
+            let badgeCount = store.badgeCount ?? 0
+            // Update badge count in payload
+            if let newRequest = request.content.mutableCopy() as? UNMutableNotificationContent {
+                newRequest.badge = NSNumber(value: badgeCount + 1)
+                store.badgeCount = badgeCount + 1
+                contentHandler(newRequest)
```

Comment:
this means `contentHandler` will be called twice I think? once here and once by the `RichPushRequestHandler`. should we instead be mutating the request that gets passed along, or handling this within the `RichPushRequestHandler`

Reply:
I agree in that the logic that modifies the badge count here should be moved into `RichPushRequestHandler`/`RichPushRequest`. 

This extension function built on top of `NotificationService`'s job is to delegate work to other functions, not perform logic in itself. 

I am realizing now that we don't document that very well. How we use extension functions for convenience and not as a way to perform logic. Code gets re-used better and is easier to test when the logic is in a SDK class instead of in an extension function.